Initiate some documentation

allyhawkins opened the pull request:
Closes #19 

This PR starts some documentation for the repo. 

- I updated the main README file to include a list of the figures and tables, how to generate figures, the contents of the `sample-info` and `palettes` folders, and some brief instructions for using `renv`. 
- For the instructions on generating figures, I added some info about what folders store what to the main README, but then created a separate README in the `scripts` directory. In that README I broke it up into figure set up and then actually generating the figures and listed each script. 
- The instructions for using `renv` I copied directly from `sc-data-integration`. They probably don't need to be this detailed, but sometimes its helpful to have the reminders? 

The other thing I did was take the time to standardize how we are naming the scripts for generating each figure. I updated them to be in the following format: `<figure/table>_<letter/number>-<figure description with words separated by underscores>.R` 

I'm sure these docs will continue to change and get updated, but let me know if there is anything obvious I missed here.

sjspielman commented on README.md:
```diff
@@ -1 +1,65 @@
+
+**Figure 1**
+
+A. Bar chart summarizing the types of diagnoses found on the Portal.
+B. Bar chart summarizing the types of modalities found on the Portal.
```
Sadly this isn't coming out as bullet list. Might need to do:

```
- Panel A: ...
- Panel B: ...
```

allyhawkins replied:
I wasn't actually trying to make a bulleted list, but I can.

sjspielman replied:
I dont think it needs to be bullet, just is hard to see now: `A. Bar chart summarizing the types of diagnoses found on the Portal. B. Bar chart summarizing the types of modalities found on the Portal.`

Maybe a semicolon!

allyhawkins replied:
Oh you are right, I forgot it wouldn't just magically be on a new line.

sjspielman commented on README.md:
```diff
@@ -1 +1,65 @@
+
+## Generating figures
+
+The `figures` and `tables` folders contain the most up-to-date version of each of the figures and tables.
```
```suggestion
The `figures` and `tables` folders contain the most up-to-date version of each of the figures and tables, respectively.
```

sjspielman commented on README.md:
```diff
@@ -1 +1,65 @@
+
+Each time you install or use new packages, you will want to run `renv::snapshot()` to update the `renv.lock` file with any added package and dependencies necessary to run the analyses and scripts in this repo.
+
+If there are dependencies you want to include that are not captured automatically by `renv::snapshot()`, add them to `components/dependencies.R` with a call to `library()` and an explanatory comment.
```
We don't have a `components` directory, let alone `components/dependencies.R`. Do we need one though? We could either:

- establish a blank file in the folder
- delete this part of docs and add it later if/when we need to use this approach

I leave this up to you!

sjspielman commented on scripts/README.md:
```diff
@@ -0,0 +1,29 @@
+
+The `figure_setup` folder contains scripts that are required to be run _prior to_ generating figures.
+
+1. `sync-metadata.R`: This script is used to sync any metadata files found on S3 to a local folder.
+In particular, the `scpca-sample-metadata.tsv` and `scpca-library-metadata.tsv` will be stored to a folder within the root directory of this repo named `s3_files`.
+In order to generate some of the figures (see more on which figures require this script below), this script will be need to run first.
```
Worth also mentioning that while the directory is placed in the root, it's also ignored so it won't be tracked with the repo. This means the script _has_ to be run; you can't just get these files by cloning the repo and/or pulling down. You do say it needs to be run first, but since you also say it's in the repo root, I think it's worth clarifying a bit more.

sjspielman commented on scripts/README.md:
```diff
@@ -0,0 +1,29 @@
+2. `fig_1b-modality_barchart.R`: This script is used to generate Figure 1B, which includes a summary of the types of modalities found in the ScPCA portal.
+Before running this script, you must run `figure_setup/sync-metadata.R`.
+
+3. `table_s1-modality_summary.R`: This script is used to generate supplemental Table 1, which contains a summary of the types of libraries found in each project.
```
Place this in a new section `## Generating tables` as item 1, or change the header above to `## Generating figures and tables`. You choose the approach! Just keep it consistent with however you choose to organize in the root README.

Reading this list also bought up a question, do we want to wrap all these in a bash script, or two scripts? (figures & tables, or main & SI). Adding that would be a separate issue/PR!

sjspielman replied:
I was thinking that it would also make the GHA easier!

sjspielman commented on README.md:
```diff
@@ -1 +1,65 @@
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Summary of figures
```
```suggestion
## Summary of figures and tables
```

sjspielman replied:
or, make a separate section. your call.

sjspielman commented on README.md:
```diff
@@ -1 +1,65 @@
+
+Summary of libraries and types of libraries found on the Portal.
+
+## Generating figures
```
again, "and tables", or a separate section.

allyhawkins commented:
I made sure to consistently use `figures and tables` throughout. I want to keep the list in order of how I expect them to be displayed/ discussed in the paper, so I like keeping it as all one list. 

I removed the section in the `renv` about dependencies. I don't think we need that at this stage anyways. 

I added a clarifying statement about the `s3_files` being ignored. 

This should be ready for another look @sjspielman!

sjspielman commented on README.md:
```diff
@@ -1 +1,61 @@
+
+**Figure 1**
+
+A. Bar chart summarizing the types of diagnoses found on the Portal.
```
```suggestion
A. Bar chart summarizing the types of diagnoses found on the Portal; 
```

sjspielman commented on scripts/README.md:
```diff
@@ -0,0 +1,30 @@
+1. `sync-metadata.R`: This script is used to sync any metadata files found on S3 to a local folder.
+In particular, the `scpca-sample-metadata.tsv` and `scpca-library-metadata.tsv` will be stored to a folder within the root directory of this repo named `s3_files`.
+In order to generate some of the figures (see more on which figures require this script below), this script will be need to run first.
+The `s3_files` folder is ignored in the `.gitignore`, so the first time you go to generate figures you will need to run this script as these files are not available in the repo.
```
avoid "double negative"?
```suggestion
The `s3_files` folder is included in the `.gitignore`, so the first time you go to generate figures you will need to run this script as these files are not available in the repo.
```

sjspielman commented on scripts/README.md:
```diff
@@ -0,0 +1,30 @@
+
+## Generating figures and tables
+
+The following scripts can be used to generate figures:
```
```suggestion
The following scripts can be used to generate figures and tables:
```

sjspielman reviewed (approved):
LGTM! Also need to re-run `doctoc` in the READMEs, but I dont need to see again.

allyhawkins commented on README.md:
```diff
@@ -1 +1,61 @@
+
+**Figure 1**
+
+A. Bar chart summarizing the types of diagnoses found on the Portal.
+B. Bar chart summarizing the types of modalities found on the Portal.
```
```suggestion
- A. Bar chart summarizing the types of diagnoses found on the Portal.
- B. Bar chart summarizing the types of modalities found on the Portal.
```
